app/GraphQL/Ecosystem/Mutations/Notifications/NotificationsManagementMutation.php

```diff
@@ -4,27 +4,88 @@
 
 namespace App\GraphQL\Ecosystem\Mutations\Notifications;
 
+use Baka\Support\Str;
 use Illuminate\Support\Facades\Notification;
+use Kanvas\AccessControlList\Enums\RolesEnums;
+use Kanvas\Apps\Models\Apps;
+use Kanvas\Notifications\Actions\EvaluateNotificationsLogicAction;
+use Kanvas\Notifications\Actions\SendMessageNotificationsToAllFollowersAction;
+use Kanvas\Notifications\Actions\SendMessageNotificationsToOneFollowerAction;
+use Kanvas\Notifications\Repositories\NotificationTypesMessageLogicRepository;
 use Kanvas\Notifications\Templates\Blank;
+use Kanvas\Social\MessagesTypes\Repositories\MessagesTypesRepository;
+use Kanvas\Users\Models\Users;
 use Kanvas\Users\Repositories\UsersRepository;
 
 class NotificationsManagementMutation
 {
     /**
      * sendNotificationBaseOnTemplate
+     * @psalm-suppress MixedArgument
      */
     public function sendNotificationBaseOnTemplate(mixed $root, array $request): bool
     {
-        $users = UsersRepository::findUsersByIds($request['users_id']);
+        $user = auth()->user();
+        $company = $user->getCurrentCompany();
+
+        if ($user->isAn(RolesEnums::OWNER->value)) {
+            $userToNotify = UsersRepository::findUsersByIds($request['users_id']);
+        } else {
+            $userToNotify = UsersRepository::findUsersByIds($request['users_id'], $company);
+        }
+
         $notification = new Blank(
             $request['template_name'],
-            is_string($request['data']) ? json_decode($request['data']) : $request['data'], // This can have more validation like validate if is array o json
+            Str::isJson($request['data']) ? json_decode($request['data'], true) : (array) $request['data'], // This can have more validation like validate if is array o json
             $request['via'],
-            auth()->user()
+            $user
         );
-        $notification->setFromUser(auth()->user());
 
-        Notification::send($users, $notification);
+        $notification->setFromUser($user);
+        Notification::send($userToNotify, $notification);
+
         return true;
     }
+
+    /**
+     * sendNotificationByMessage
+     * @psalm-suppress MixedArgument
+     */
+    public function sendNotificationByMessage(mixed $root, array $request): bool
+    {
+        $app = app(Apps::class);
+        $user = auth()->user();
+        $message = $request['message'];
+
+        // TODO Maybe get rid of the notification_type_id on notification_types_message_logic table, not doing anything there?
+        $messageType = MessagesTypesRepository::getByVerb($message['metadata']['verb'], $app);
```

Comment:
@rwhite27 doesnt this message have any type of structure you can add to a DTO instead of mixed ?